### ISSUE TITLE
ESP32S3 configuration USB-Serial output log will unexpectedly restarted (IDFGH-5921)

### ISSUE THREAD

**Issue by dreamcmi** [espressif/esp-idf#7612]
INSTRUCTIONS
============

Before submitting a new issue, please follow the checklist and try to find the answer.

- [x] I have read the documentation [ESP-IDF Programming Guide](https://docs.espressif.com/projects/esp-idf/en/latest/) and the issue is not addressed there.
- [ ] I have updated my IDF branch (master or release) to the latest version and checked that the issue is present there.
- [ ] I have searched the issue tracker for a similar issue and not found a similar issue.

## Environment

- Development Kit:      [ESP32-S3-DevKitC-1]
- Kit version (for WroverKit/PicoKit/DevKitC): [v1]
- Module or chip used:  [ESP32-S3-WROOM-1]
- IDF version :  v4.4-dev-3042-g220590d599  
- Build System:         [idf.py]
- Compiler version : xtensa-esp32s3-elf-gcc.exe (crosstool-NG esp-2021r1) 8.4.0  
- Operating System:     [Windows]
- (Windows only) environment type: [PowerShell].
- Using an IDE?: [VSCode]
- Power Supply:         [USB]

## Problem Description

I have enabled the USB Serial/JTAG controller option in `idf.py menuconfig`, but after downloading to the chip, using the serial port to connect to the computer, the error of `rst:0x10 (RTCWDT_RTC_RST), boot:0x8 (SPI_FAST_FLASH_BOOT)` will occur, and it must be plugged in The program can run normally after usb. This will be very troublesome, because the device will not always be plugged in to run when it is running normally. How can I solve this problem? 

## Debug Logs

The following is the serial output  

```
[2021/09/28 22:43:01.0518] → ESP-ROM:esp32s3-20210327
Build:Mar 27 2021
rst:0x1 (POWERON),boot:0x8 (SPI_FAST_FLASH_BOOT)
SPIWP:0xee
mode:DIO, clock div:1
load:0x3fcd0108,len:0x1414
load:0x403b6000,len:0x8ac
load:0x403ba000,len:0x2e60
entry 0x403b61b8
I (24) boot: ESP-IDF v4.4-dev-3042-g220590d599-dirty 2nd stage bootloader
I (25) boot: compile time 22:29:31
I (25) boot: chip revision: 0
I (26) boot.esp32s3: Boot SPI Speed : 80MHz
I (29) boot.esp32s3: SPI Mode       : DIO
I (33) boot.esp32s3: SPI Flash Size : 8MB
I (37) boot: Enabling RNG early entropy source...
W (41) bootloader_random: RNG for ESP32-S3 not currently supported
I (47) boot: Partition Table:
I (50) boot: ## Label            Usage          Type ST Offset   Length
I (56) boot:  0 nvs              WiFi data        01 02 00009000 00005000
I (63) boot:  1 otadata          OTA data         01 00 0000e000 00002000
I (69) boot:  2 app0             OTA app          00 10 00010000 00330000
I (76) boot:  3 app1             OTA app          00 11 00340000 00330000
I (82) boot:  4 spiffs           Unknown data     01 82 00670000 00190000
I (89) boot: End of partition table
I (92) esp_image: segment 0: paddr=00010020 vaddr=3c070020 size=122c0h ( 74432) map
I (113) esp_image: segment 1: paddr=000222e8 vaddr=3fc93f00 size=03f4ch ( 16204) load
I (117) esp_image: segment 2: paddr=0002623c vaddr=40374000 size=09ddch ( 40412) load
I (126) esp_image: segment 3: paddr=00030020 vaddr=42000020 size=65b00h (416512) map
I (201) esp_image: segment 4: paddr=00095b28 vaddr=4037dddc size=06118h ( 24856) load
I (207) esp_image: segment 5: paddr=0009bc48 vaddr=50000000 size=00010h (    16) load
I (213) boot: Loaded app from partition at offset 0x10000
I (213) boot: Disabling RNG early entropy source...
W (214) bootloader_random: RNG for ESP32-S3 not currently supported
I (221) opi psram: vendor id : 0x0d (AP)
I (224) opi psram: dev id    : 0x02 (generation 3)
I (229) opi psram: density   : 0x03 (64 Mbit)
I (233) opi psram: good-die  : 0x01 (Pass)
I (236) opi psram: Latency   : 0x01 (Fixed)
I (240) opi psram: VCC       : 0x01 (3V)
I (244) opi psram: SRF       : 0x01 (Fast Refresh)
I (249) opi psram: BurstType : 0x01 (Hybrid Wrap)
I (253) opi psram: BurstLen  : 0x01 (32 Byte)
I (257) opi psram: Readlatency  : 0x02 (10 cycles@Fixed)
I (262) opi psram: DriveStrength: 0x00 (1/1)
I (266) spiram: Found 64MBit SPI RAM device
I (270) spiram: SPI RAM mode: sram 40m
I (273) spiram: PSRAM initialized, cache is in normal (1-core) mode.
I (280) cpu_start: Pro cpu up.
I (282) cpu_start: Starting app cpu, entry point is 0x40375220

[2021/09/28 22:43:12.6252] → ESP-ROM:esp32s3-20210327
Build:Mar 27 2021
rst:0x10 (RTCWDT_RTC_RST),boot:0x8 (SPI_FAST_FLASH_BOOT)
SPIWP:0xee
mode:DIO, clock div:1
load:0x3fcd0108,len:0x1414
load:0x403b6000,len:0x8ac
load:0x403ba000,len:0x2e60
entry 0x403b61b8
I (25) boot: ESP-IDF v4.4-dev-3042-g220590d599-dirty 2nd stage bootloader
I (25) boot: compile time 22:29:31
I (25) boot: chip revision: 0
I (26) boot.esp32s3: Boot SPI Speed : 80MHz
I (30) boot.esp32s3: SPI Mode       : DIO
I (34) boot.esp32s3: SPI Flash Size : 8MB
I (38) boot: Enabling RNG early entropy source...
W (42) bootloader_random: RNG for ESP32-S3 not currently supported
I (48) boot: Partition Table:
I (51) boot: ## Label            Usage          Type ST Offset   Length
I (57) boot:  0 nvs              WiFi data        01 02 00009000 00005000
I (64) boot:  1 otadata          OTA data         01 00 0000e000 00002000
I (70) boot:  2 app0             OTA app          00 10 00010000 00330000
I (77) boot:  3 app1             OTA app          00 11 00340000 00330000
I (83) boot:  4 spiffs           Unknown data     01 82 00670000 00190000
I (90) boot: End of partition table
I (93) esp_image: segment 0: paddr=00010020 vaddr=3c070020 size=122c0h ( 74432) map
I (114) esp_image: segment 1: paddr=000222e8 vaddr=3fc93f00 size=03f4ch ( 16204) load
I (117) esp_image: segment 2: paddr=0002623c vaddr=40374000 size=09ddch ( 40412) load
I (126) esp_image: segment 3: paddr=00030020 vaddr=42000020 size=65b00h (416512) map
I (202) esp_image: segment 4: paddr=00095b28 vaddr=4037dddc size=06118h ( 24856) load
I (207) esp_image: segment 5: paddr=0009bc48 vaddr=50000000 size=00010h (    16) load
I (214) boot: Loaded app from partition at offset 0x10000
I (214) boot: Disabling RNG early entropy source...
W (215) bootloader_random: RNG for ESP32-S3 not currently supported
I (221) opi psram: vendor id : 0x0d (AP)
I (225) opi psram: dev id    : 0x02 (generation 3)
I (229) opi psram: density   : 0x03 (64 Mbit)
I (233) opi psram: good-die  : 0x01 (Pass)
I (237) opi psram: Latency   : 0x01 (Fixed)
I (241) opi psram: VCC       : 0x01 (3V)
I (245) opi psram: SRF       : 0x01 (Fast Refresh)
I (249) opi psram: BurstType : 0x01 (Hybrid Wrap)
I (254) opi psram: BurstLen  : 0x01 (32 Byte)
I (258) opi psram: Readlatency  : 0x02 (10 cycles@Fixed)
I (263) opi psram: DriveStrength: 0x00 (1/1)
I (267) spiram: Found 64MBit SPI RAM device
I (271) spiram: SPI RAM mode: sram 40m
I (274) spiram: PSRAM initialized, cache is in normal (1-core) mode.
I (280) cpu_start: Pro cpu up.
I (283) cpu_start: Starting app cpu, entry point is 0x40375220

````
The following is the project file：    
[espnow.zip](https://github.com/espressif/esp-idf/files/7245437/espnow.zip)

**Comment by mythbuster5** [comment]
Hi, Cheng.

Firstly, is it possible to try usb_serial port on linux? I'm not sure that usb_serial port on windows can work properly.

Secondly, are you pretty sure that the issue is caused by usb_serial. Two things can help you check.
-(1) using uart port to see whether there is still a restarting issue or not.
-(2) Run hello_world example with usb_serial port.

**Comment by dreamcmi** [comment]
Thanks to answer:
(1) I use the usb_serial on windows
(2) I try to run the hello_world example with usb_serial port,but it still report error.  
```
[2021/09/29 14:56:17.8737] → ESP-ROM:esp32s3-20210327
Build:Mar 27 2021
rst:0x1 (POWERON),boot:0x8 (SPI_FAST_FLASH_BOOT)
SPIWP:0xee
mode:DIO, clock div:1
load:0x3fcd0108,len:0x1718
load:0x403b6000,len:0x8ac
load:0x403ba000,len:0x2e60
entry 0x403b61b8
[0;32mI (24) boot: ESP-IDF v4.4-dev-3042-g220590d599-dirty 2nd stage bootloader[0m
[0;32mI (25) boot: compile time 14:53:50[0m
[0;32mI (25) boot: chip revision: 0[0m
[0;32mI (28) boot.esp32s3: Boot SPI Speed : 80MHz[0m
[0;32mI (33) boot.esp32s3: SPI Mode       : DIO[0m
[0;32mI (38) boot.esp32s3: SPI Flash Size : 2MB[0m
[0;32mI (43) boot: Enabling RNG early entropy source...[0m
[0;33mW (48) bootloader_random: RNG for ESP32-S3 not currently supported[0m
[0;32mI (55) boot: Partition Table:[0m
[0;32mI (59) boot: ## Label            Usage          Type ST Offset   Length[0m
[0;32mI (66) boot:  0 nvs              WiFi data        01 02 00009000 00006000[0m
[0;32mI (73) boot:  1 phy_init         RF data          01 01 0000f000 00001000[0m
[0;32mI (81) boot:  2 factory          factory app      00 00 00010000 00100000[0m
[0;32mI (88) boot: End of partition table[0m
[0;32mI (92) esp_image: segment 0: paddr=00010020 vaddr=3c020020 size=06fc0h ( 28608) map[0m
[0;32mI (106) esp_image: segment 1: paddr=00016fe8 vaddr=3fc90640 size=025d4h (  9684) load[0m
[0;32mI (111) esp_image: segment 2: paddr=000195c4 vaddr=40374000 size=06a54h ( 27220) load[0m
[0;32mI (124) esp_image: segment 3: paddr=00020020 vaddr=42000020 size=14604h ( 83460) map[0m
[0;32mI (141) esp_image: segment 4: paddr=0003462c vaddr=4037aa54 size=05be8h ( 23528) load[0m
[0;32mI (147) esp_image: segment 5: paddr=0003a21c vaddr=50000000 size=00010h (    16) load[0m
[0;32mI (152) boot: Loaded app from partition at offset 0x10000[0m
[0;32mI (153) boot: Disabling RNG early entropy source...[0m
[0;33mW (159) bootloader_random: RNG for ESP32-S3 not currently supported[0m
[0;32mI (166) opi psram: vendor id : 0x0d (AP)[0m
[0;32mI (170) opi psram: dev id    : 0x02 (generation 3)[0m
[0;32mI (176) opi psram: density   : 0x03 (64 Mbit)[0m
[0;32mI (181) opi psram: good-die  : 0x01 (Pass)[0m
[0;32mI (185) opi psram: Latency   : 0x01 (Fixed)[0m
[0;32mI (190) opi psram: VCC       : 0x01 (3V)[0m
[0;32mI (195) opi psram: SRF       : 0x01 (Fast Refresh)[0m
[0;32mI (200) opi psram: BurstType : 0x01 (Hybrid Wrap)[0m
[0;32mI (206) opi psram: BurstLen  : 0x01 (32 Byte)[0m
[0;32mI (211) opi psram: Readlatency  : 0x02 (10 cycles@Fixed)[0m
[0;32mI (217) opi psram: DriveStrength: 0x00 (1/1)[0m
[0;32mI (222) spiram: Found 64MBit SPI RAM device[0m
[0;32mI (227) spiram: SPI RAM mode: sram 40m[0m
[0;32mI (231) spiram: PSRAM initialized, cache is in normal (1-core) mode.[0m
[0;32mI (238) cpu_start: Pro cpu up.[0m
[0;32mI (242) cpu_start: Starting app cpu, entry point is 0x40375124[0m

[2021/09/29 14:56:29.2805] → ESP-ROM:esp32s3-20210327
Build:Mar 27 2021
rst:0x10 (RTCWDT_RTC_RST),boot:0x8 (SPI_FAST_FLASH_BOOT)
SPIWP:0xee
mode:DIO, clock div:1
load:0x3fcd0108,len:0x1718
load:0x403b6000,len:0x8ac
load:0x403ba000,len:0x2e60
entry 0x403b61b8
[0;32mI (25) boot: ESP-IDF v4.4-dev-3042-g220590d599-dirty 2nd stage bootloader[0m
[0;32mI (25) boot: compile time 14:53:50[0m
[0;32mI (26) boot: chip revision: 0[0m
[0;32mI (29) boot.esp32s3: Boot SPI Speed : 80MHz[0m
[0;32mI (34) boot.esp32s3: SPI Mode       : DIO[0m
[0;32mI (39) boot.esp32s3: SPI Flash Size : 2MB[0m
[0;32mI (43) boot: Enabling RNG early entropy source...[0m
[0;33mW (49) bootloader_random: RNG for ESP32-S3 not currently supported[0m
[0;32mI (56) boot: Partition Table:[0m
[0;32mI (59) boot: ## Label            Usage          Type ST Offset   Length[0m
[0;32mI (67) boot:  0 nvs              WiFi data        01 02 00009000 00006000[0m
[0;32mI (74) boot:  1 phy_init         RF data          01 01 0000f000 00001000[0m
[0;32mI (81) boot:  2 factory          factory app      00 00 00010000 00100000[0m
[0;32mI (89) boot: End of partition table[0m
[0;32mI (93) esp_image: segment 0: paddr=00010020 vaddr=3c020020 size=06fc0h ( 28608) map[0m
[0;32mI (107) esp_image: segment 1: paddr=00016fe8 vaddr=3fc90640 size=025d4h (  9684) load[0m
[0;32mI (112) esp_image: segment 2: paddr=000195c4 vaddr=40374000 size=06a54h ( 27220) load[0m
[0;32mI (124) esp_image: segment 3: paddr=00020020 vaddr=42000020 size=14604h ( 83460) map[0m
[0;32mI (142) esp_image: segment 4: paddr=0003462c vaddr=4037aa54 size=05be8h ( 23528) load[0m
[0;32mI (147) esp_image: segment 5: paddr=0003a21c vaddr=50000000 size=00010h (    16) load[0m
[0;32mI (153) boot: Loaded app from partition at offset 0x10000[0m
[0;32mI (154) boot: Disabling RNG early entropy source...[0m
[0;33mW (159) bootloader_random: RNG for ESP32-S3 not currently supported[0m
[0;32mI (166) opi psram: vendor id : 0x0d (AP)[0m
[0;32mI (171) opi psram: dev id    : 0x02 (generation 3)[0m
[0;32mI (176) opi psram: density   : 0x03 (64 Mbit)[0m
[0;32mI (181) opi psram: good-die  : 0x01 (Pass)[0m
[0;32mI (186) opi psram: Latency   : 0x01 (Fixed)[0m
[0;32mI (191) opi psram: VCC       : 0x01 (3V)[0m
[0;32mI (196) opi psram: SRF       : 0x01 (Fast Refresh)[0m
[0;32mI (201) opi psram: BurstType : 0x01 (Hybrid Wrap)[0m
[0;32mI (206) opi psram: BurstLen  : 0x01 (32 Byte)[0m
[0;32mI (211) opi psram: Readlatency  : 0x02 (10 cycles@Fixed)[0m
[0;32mI (217) opi psram: DriveStrength: 0x00 (1/1)[0m
[0;32mI (222) spiram: Found 64MBit SPI RAM device[0m
[0;32mI (227) spiram: SPI RAM mode: sram 40m[0m
[0;32mI (232) spiram: PSRAM initialized, cache is in normal (1-core) mode.[0m
[0;32mI (239) cpu_start: Pro cpu up.[0m
[0;32mI (242) cpu_start: Starting app cpu, entry point is 0x40375124[0m

[2021/09/29 14:56:40.7425] → ESP-ROM:esp32s3-20210327
Build:Mar 27 2021
rst:0x10 (RTCWDT_RTC_RST),boot:0x8 (SPI_FAST_FLASH_BOOT)
SPIWP:0xee
mode:DIO, clock div:1
load:0x3fcd0108,len:0x1718
load:0x403b6000,len:0x8ac
load:0x403ba000,len:0x2e60
entry 0x403b61b8
[0;32mI (25) boot: ESP-IDF v4.4-dev-3042-g220590d599-dirty 2nd stage bootloader[0m
[0;32mI (25) boot: compile time 14:53:50[0m
[0;32mI (26) boot: chip revision: 0[0m
[0;32mI (29) boot.esp32s3: Boot SPI Speed : 80MHz[0m
[0;32mI (34) boot.esp32s3: SPI Mode       : DIO[0m
[0;32mI (39) boot.esp32s3: SPI Flash Size : 2MB[0m
[0;32mI (43) boot: Enabling RNG early entropy source...[0m
[0;33mW (49) bootloader_random: RNG for ESP32-S3 not currently supported[0m
[0;32mI (56) boot: Partition Table:[0m
[0;32mI (59) boot: ## Label            Usage          Type ST Offset   Length[0m
[0;32mI (67) boot:  0 nvs              WiFi data        01 02 00009000 00006000[0m
[0;32mI (74) boot:  1 phy_init         RF data          01 01 0000f000 00001000[0m
[0;32mI (81) boot:  2 factory          factory app      00 00 00010000 00100000[0m
[0;32mI (89) boot: End of partition table[0m
[0;32mI (93) esp_image: segment 0: paddr=00010020 vaddr=3c020020 size=06fc0h ( 28608) map[0m
[0;32mI (107) esp_image: segment 1: paddr=00016fe8 vaddr=3fc90640 size=025d4h (  9684) load[0m
[0;32mI (112) esp_image: segment 2: paddr=000195c4 vaddr=40374000 size=06a54h ( 27220) load[0m
[0;32mI (124) esp_image: segment 3: paddr=00020020 vaddr=42000020 size=14604h ( 83460) map[0m
[0;32mI (142) esp_image: segment 4: paddr=0003462c vaddr=4037aa54 size=05be8h ( 23528) load[0m
[0;32mI (147) esp_image: segment 5: paddr=0003a21c vaddr=50000000 size=00010h (    16) load[0m
[0;32mI (153) boot: Loaded app from partition at offset 0x10000[0m
[0;32mI (154) boot: Disabling RNG early entropy source...[0m
[0;33mW (159) bootloader_random: RNG for ESP32-S3 not currently supported[0m
[0;32mI (166) opi psram: vendor id : 0x0d (AP)[0m
[0;32mI (171) opi psram: dev id    : 0x02 (generation 3)[0m
[0;32mI (176) opi psram: density   : 0x03 (64 Mbit)[0m
[0;32mI (181) opi psram: good-die  : 0x01 (Pass)[0m
[0;32mI (186) opi psram: Latency   : 0x01 (Fixed)[0m
[0;32mI (191) opi psram: VCC       : 0x01 (3V)[0m
[0;32mI (196) opi psram: SRF       : 0x01 (Fast Refresh)[0m
[0;32mI (201) opi psram: BurstType : 0x01 (Hybrid Wrap)[0m
[0;32mI (206) opi psram: BurstLen  : 0x01 (32 Byte)[0m
[0;32mI (211) opi psram: Readlatency  : 0x02 (10 cycles@Fixed)[0m
[0;32mI (217) opi psram: DriveStrength: 0x00 (1/1)[0m
[0;32mI (222) spiram: Found 64MBit SPI RAM device[0m
[0;32mI (227) spiram: SPI RAM mode: sram 40m[0m
[0;32mI (232) spiram: PSRAM initialized, cache is in normal (1-core) mode.[0m
[0;32mI (239) cpu_start: Pro cpu up.[0m
[0;32mI (242) cpu_start: Starting app cpu, entry point is 0x40375124[0m
```
[sdkconfig.zip](https://github.com/espressif/esp-idf/files/7249560/sdkconfig.zip)
 In the end,i did not think the problem was windows or linux problem ,because i did not plug in the usb and just used the serial port to view the log,thanks.

**Comment by meltdown03** [comment]
I get the same issue on linux. If I set the console to use USB/JTAG and don't use the USB port (plug it in to the UART port instead), I get constant bootloops. I also don't get any console output from the program once it's running. I'm running the ESP-Rainmaker led_light example on the ESP32-S3-DevkitC-1, 8MB, no PSRAM version

**Comment by dreamcmi** [comment]
> I get the same issue on linux. If I set the console to use USB/JTAG and don't use the USB port (plug it in to the UART port instead), I get constant bootloops. I also don't get any console output from the program once it's running. I'm running the ESP-Rainmaker led_light example on the ESP32-S3-DevkitC-1, 8MB, no PSRAM version

So I think the problem is on esp-idf, and it has nothing to do with the OS used, but the official has never responded,  I don’t know how to solve the problem.

**Comment by mythbuster5** [comment]
> I get the same issue on linux. If I set the console to use USB/JTAG and don't use the USB port (plug it in to the UART port instead), I get constant bootloops. I also don't get any console output from the program once it's running. I'm running the ESP-Rainmaker led_light example on the ESP32-S3-DevkitC-1, 8MB, no PSRAM version

Could you please show me the port on linux you are using. with `ls -l /dev/tty*`

**Comment by igrr** [comment]
This should be fixed by 4f5ab6be.
